### ISSUE TITLE
[1.2.0] Subquery crashes InfluxDB and host

### ISSUE THREAD

**Issue by neographikal** [influxdata/influxdb#8021]
### Bug report

__System info:__ InfluxDB 1.2 on Ubuntu 16.04.2 LTS in VMWare Guest

__Steps to reproduce:__

I've got a simple database with a series of timestamps with corresponding kwh (float). 

```
select * from splitheatpump LIMIT 10
name: splitheatpump
time                value
----                -----
1486339260000000000 0.0004999999999988347
1486339320000000000 0.0004999999999988347
1486339380000000000 0.0005000000000023874
1486339440000000000 0.0004999999999988347
1486339500000000000 0.0004999999999988347
1486339560000000000 0.0005000000000023874
1486339620000000000 0.0004999999999988347
1486339680000000000 0.0004999999999988347
1486339740000000000 0.0005000000000023874
1486339800000000000 0.0004999999999988347

show series
key
---
solarpanels
splitheatpump
```

Running this query gives an instant result as I expect it:

```
select (ELAPSED("value")* -1)/1000000000 AS etime  from splitheatpump order by time desc limit 1
name: splitheatpump
time                etime
----                -----
1486943880000000000 60
```

Wrap it as a subquery like this:

`SELECT * from (select (ELAPSED("value")* -1)/1000000000 AS etime  from splitheatpump order by time desc limit 1)`

InfluxDB immediatly eats all mem and cpu it can get.  System crashes.

__Expected behavior:__ Give results and do not crash

__Actual behavior:__ High cpu/mem usage, hangs system completely

__Additional info:__ Default influxdb install on Ubuntu

**Comment by neographikal** [comment]
Looks a bit like #8023

**Comment by jwilder** [comment]
Not reproducible w/ 7c23d6869cbae43c2b562346b9ee45d4b2d8ab01.  This should be fixed when 1.2.1 is released.
